app/src/main/cpp/skyline/vfs/directory.h

```diff
@@ -19,17 +22,19 @@ namespace skyline::vfs {
         struct Entry {
             std::string name;
             EntryType type;
+            size_t size; //!< 0 if a directory
         };
 
         /**
          * @brief A descriptor for what will be returned when reading a directories contents
+         * @note MUST match HOS
```

Comment:
This doesn't fit the rest of the codebase, a lot of structures need to match HOS and we don't explicitly mark them as such

Reply:
Well this is part of vfs, other vfs structs don't need to match hos (see entry below). If you have a list of structs in other subsystems that need to match hos tell me and I'll add notes

Reply:
This applies to a lot of subsystems, it's readily apparent when structures (Esp. unions) are meant for the guest. This shouldn't just set aside that entire convention and need one here.

Reply:
Not really, anyone could assume it's not meant for the guest. Please tell me what subsystems you speak of so I can fix convention

Reply:
The convention doesn't need fixing but essentially all services, kernel objects and GPU.

Reply:
all of them are obvious and all structs there are almost always guest, that isn't the case for vfs where most aren't for HOS

Reply:
Are you saying this isn't obvious with it being a union with a `raw` value and a bitfield struct, if you assume this is not for guest then I'm not sure what to say

Reply:
it's definitely not obvious for EntryType, it's literally an enum